packages/adapters/eth/src/fee_api_helpers.rs

```diff
@@ -0,0 +1,561 @@
+
+    #[test]
+    fn test_unpack_fee_history_multiple_blocks() {
+        // given
+        let fees = FeeHistory {
+            oldest_block: 700,
+            base_fee_per_gas: vec![100, 200, 300, 400], // number_of_blocks =3
+            base_fee_per_blob_gas: vec![150, 250, 350, 450],
+            reward: Some(vec![vec![10], vec![20], vec![30]]),
+            ..Default::default()
+        };
+
+        // when
+        let result = unpack_fee_history(fees);
+
+        // then
+        let expected = vec![
+            FeesAtHeight {
+                height: 700,
+                fees: Fees {
+                    base_fee_per_gas: 100.try_into().unwrap(),
+                    reward: 10.try_into().unwrap(),
+                    base_fee_per_blob_gas: 150.try_into().unwrap(),
+                },
+            },
+            FeesAtHeight {
+                height: 701,
+                fees: Fees {
+                    base_fee_per_gas: 200.try_into().unwrap(),
+                    reward: 20.try_into().unwrap(),
+                    base_fee_per_blob_gas: 250.try_into().unwrap(),
+                },
+            },
+            FeesAtHeight {
+                height: 702,
+                fees: Fees {
+                    base_fee_per_gas: 300.try_into().unwrap(),
+                    reward: 30.try_into().unwrap(),
+                    base_fee_per_blob_gas: 350.try_into().unwrap(),
+                },
+            },
+        ];
+        assert_eq!(
+            result.unwrap(),
+            expected,
+            "Expected three BlockFees entries for three blocks"
+        );
+    }
+
+    #[test]
+    fn test_unpack_fee_history_large_values() {
+        // given
+        let fees = FeeHistory {
+            oldest_block: u64::MAX - 2,
+            base_fee_per_gas: vec![u128::MAX - 2, u128::MAX - 1, u128::MAX],
+            base_fee_per_blob_gas: vec![u128::MAX - 3, u128::MAX - 2, u128::MAX - 1],
+            reward: Some(vec![vec![u128::MAX - 4], vec![u128::MAX - 3]]),
+            ..Default::default()
+        };
+
+        // when
+        let result = unpack_fee_history(fees.clone());
+
+        // then
+        let expected = vec![
+            FeesAtHeight {
+                height: u64::MAX - 2,
+                fees: Fees {
+                    base_fee_per_gas: u128::MAX - 2,
+                    reward: u128::MAX - 4,
+                    base_fee_per_blob_gas: u128::MAX - 3,
+                },
+            },
+            FeesAtHeight {
+                height: u64::MAX - 1,
+                fees: Fees {
+                    base_fee_per_gas: u128::MAX - 1,
+                    reward: u128::MAX - 3,
+                    base_fee_per_blob_gas: u128::MAX - 2,
+                },
+            },
+        ];
+        assert_eq!(
+            result.unwrap(),
+            expected,
+            "Expected BlockFees entries with large u64 values"
+        );
+    }
+
+    #[test]
+    fn test_unpack_fee_history_full_range_chunk() {
```

Comment:
and this vs. "test_unpack_fee_history_multiple_blocks"

Reply:
[ad1a711](https://github.com/FuelLabs/fuel-block-committer/commit/ad1a7117d56fbec13daa05faf8631bbc29c6cc39)